src/main/java/edu/ie3/datamodel/utils/ContainerUtils.java

```diff
@@ -418,4 +418,236 @@ public static JointGridContainer combineToJointGrid(
+
+    // build the updated 2w transformer with nodeA as slack and add it to our new set
+    Set<Transformer2WInput> newTrafos2w =
+        subGridContainer.getRawGrid().getTransformer2Ws().stream()
+            .map(
+                oldTrafo2w ->
+                    copyTrafo2WUpdateNodeA(
+                        oldTrafo2w, oldToNewTrafo2WNodes.get(oldTrafo2w.getNodeA())))
+            .collect(Collectors.toSet());
+
+    // update node input graphics (2 winding transformers only)
+    /// map old to new
+    Map<NodeGraphicInput, NodeGraphicInput> oldToNewNodeGraphics =
+        subGridContainer.getGraphics().getNodeGraphics().stream()
+            .filter(nodeGraphic -> oldToNewTrafo2WNodes.containsKey(nodeGraphic.getNode()))
+            .map(
+                oldNodeGraphic ->
+                    new AbstractMap.SimpleEntry<>(
+                        oldNodeGraphic,
+                        copyNodeGraphic(
+                            oldNodeGraphic, oldToNewTrafo2WNodes.get(oldNodeGraphic.getNode()))))
+            .collect(
+                Collectors.toMap(
+                    AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+
+    /// remove old node graphics, add new ones
+    Set<NodeGraphicInput> newNodeGraphics =
+        Stream.concat(
+                // filter old ones
+                subGridContainer.getGraphics().getNodeGraphics().stream()
+                    .filter(nodeGraphic -> !oldToNewNodeGraphics.containsKey(nodeGraphic)),
+                // add the new trafo2w ones
+                oldToNewNodeGraphics.values().stream())
+            .collect(Collectors.toSet());
+
+    // update nodes in raw grid by removing the old 2 winding transformer nodes and add all new ones
+    Set<NodeInput> newNodes =
+        Stream.concat(
+                // filter the old ones (trafo2w, trafo3wToBeRemoved)
+                subGridContainer.getRawGrid().getNodes().stream()
+                    .filter(node -> !oldToNewTrafo2WNodes.containsKey(node))
+                    .filter(node -> !oldToNewTrafo3WNodes.containsKey(node)),
+                // add the new ones (trafo2w, trafo3w and updated trafo3w nodeA (previous slacks))
+                Stream.concat(
+                    oldToNewTrafo2WNodes.values().stream(),
+                    Stream.concat(
+                        newTrafos3w.values().stream(), oldToNewTrafo3WNodes.values().stream())))
+            .collect(Collectors.toSet());
+
+    return new SubGridContainer(
+        subGridContainer.getGridName(),
+        subGridContainer.getSubnet(),
+        rawGridCopy4Slacks(
+            subGridContainer.getRawGrid(), newNodes, newTrafos2w, newTrafos3w.keySet()),
+        subGridContainer.getSystemParticipants(),
+        graphicsCopy4Slacks(subGridContainer.getGraphics(), newNodeGraphics));
+  }
+
+  private static Transformer3WInput copyTrafo3WWithInternalSlack(
```

Comment:
```suggestion
  private static Transformer3WInput copyTransformer3WInputWithInternalSlack(
```
Maybe to stick a little more to the model class? Would you mind adding JavaDoc?